src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/orchestrator/network/_network_orchestrator.py

```diff
@@ -0,0 +1,57 @@
+from ...constants import Constant
+from ...orchestrator._orchestrator import Orchestrator
+import ipaddress
+from avs.avsarconboarder.exception import InvalidInputError
+
+class NetworkOrchestrator(Orchestrator):
+
+    def validate_ip_address(self, ip_addr):
+        try:
+            ipaddress.ip_address(ip_addr)
+        except:
+            raise InvalidInputError("Invalid ip addresses provided")
+
+    def validate_static_ip_cidr_block(self, segment_ip_cidr, version):
+        gateway_ip, cidr = self.get_gateway_address_cidr_from_network_addr(segment_ip_cidr)
+        self.validate_ip_address(gateway_ip)
+        
+        if version == Constant.NEW_CONFIG_VERSION:
+            if int(cidr) != 28:
+                raise InvalidInputError("Invalid segment block size provided, Please provide a /28 address")
+            
+            subnet_first_ip_addr = ipaddress.IPv4Address(gateway_ip) - 1
+            try:
+                ipaddress.ip_network('{}/{}'.format(subnet_first_ip_addr,cidr))
+            except:
+                raise InvalidInputError("Invalid gateway ip provided, please provide a segment with first gateway ip")          
+        
+    def get_gateway_address_cidr_from_network_addr(self, segment_ip_cidr):
```

Comment:
>get_gateway_address_cidr_from_network_addr [](http://example.com/codeflow?start=8&length=42)

nit: any reason to not use split ? It would simplify the entire block.

Reply:
We can do but all cases may not be covered.
for example - ip cidr given is 127.1.1.1/28/3. then if we are splitting with '/'
['127.1.1.1','28','3'], so even with a simple split we may have other corner cases. I know I am talking about very extreme corner cases :).

Reply:
For my knowledge, is this (127.1.1.1/28/3) a valid cidr? What does this denote?

Reply:
It is not a valid cidr, but if customer is prone to error and might give such value and we deploy arc appliance with this. We may fail lately in arc validation. But given customer doesn't do such typo or rookie mistake, we can always use the simple split method as you suggested.

Reply:
Used the split fuinction. this looks more cleaner. Also put a check of it is a valid format or not.